PictureApp.API.Services/FollowerService.cs

```diff
@@ -0,0 +1,107 @@
+        private readonly IRepository<User> _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public FollowerService(
+            IUserService userService,
+            IRepository<UserFollower> repository, 
+            IRepository<User> userRepository, 
+            IUnitOfWork unitOfWork, 
+            IMapper mapper)
+        {
+            _userService = userService;
+            _repository = repository;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<UsersListWithFollowersForExploreDto>> GetAllWithFollowers(int userId)
+        {   
+            var usersFollower = await _repository.FindAsync(x => x.FollowerId == userId);
+            var users = await _userRepository.GetAllAsync();
+
+            var usersWithFollowersToReturn = users.Select(user => 
+                {
+                    var mappedUser = _mapper.Map<UsersListWithFollowersForExploreDto>(user);
+                    
+                    mappedUser.IsFollowerForCurrentUser = usersFollower.Any(userFollower => userFollower.FolloweeId == user.Id);
+     
+                    return mappedUser;
+                }).ToList();
+            
+            return usersWithFollowersToReturn;
+        }
+
+        public async Task SetUpFollower(int userId, int recipientId)
+        {
+            await ValidateUser(userId, recipientId);
+
+            if (!await _repository.AnyAsync(
+                u => u.FollowerId == userId && u.FolloweeId == recipientId))
+            {
+                var follower = new UserFollower
+                {
+                    FollowerId = userId,
+                    FolloweeId = recipientId
+                };
+
+                await _repository.AddAsync(follower);
+                await _unitOfWork.CompleteAsync();
+            }
+            else 
+            {
+                ////log if user arleady follow choosen recipient
+            }
+        }
+
+        public async Task SetUpUnfollower(int userId, int recipientId)
+        {
+            await ValidateUser(userId, recipientId);
+
+            var userFollower = await _repository.FirstOrDefaultAsync(
+                u => u.FollowerId == userId && u.FolloweeId == recipientId);
+
+            if (userFollower != null)
+            {
+                _repository.Delete(userFollower);
+                await _unitOfWork.CompleteAsync();
+            }
+            else
+            {
+                ////log if user arleady unfollow choosen recipient
+            }
+        }
+
+        private async Task ValidateUser(int userId, int recipientId)
+        {
+            var user = await _userService.GetUser(userId);
+            var recpitent = await _userService.GetUser(recipientId);
+
+            if (user == null)
+                throw new NotAuthorizedException($"user by {userId} not found");
```

Comment:
Why for lack of user we use different exception?

Reply:
fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440